plenum/server/replicas.py

```diff
@@ -132,6 +132,66 @@ def register_new_ledger(self, ledger_id):
         for replica in self._replicas:
             replica.register_ledger(ledger_id)
 
+    def set_default_handlers(self):
+        self._monitor.unordered_requests_handlers.append(
+            self.unordered_request_handler_logging)
+
+    def unordered_request_handler_logging(self, unordereds):
+        replica = self._master_replica
+        for unordered in unordereds:
+            req, duration = unordered
+            reqId = req[1]
+
+            # get ppSeqNo and viewNo
+            preprepares = replica.sentPrePrepares if replica.isPrimary else replica.prePrepares
+            ppSeqNo = None
+            viewNo = None
+            for key in preprepares:
+                if any([req[1] == reqId for req in preprepares[key].reqIdr]):
```

Comment:
uniq identifier for the request is a tuple (sender, reqid), so it is better to compare all the fields from the reqIdr

Reply:
Done